modules/stochastic_tools/src/trainers/GaussianProcessTrainer.C

```diff
@@ -100,7 +100,9 @@ GaussianProcessTrainer::GaussianProcessTrainer(const InputParameters & parameter
     if (_sampler.getNumberOfRows() < _optimization_opts.batch_size)
       paramError("batch_size", "Batch size cannot be greater than the training data set size.");
 
-  std::vector<std::string> tune_parameters(getParam<std::vector<std::string>>("tune_parameters"));
+  std::vector<std::string> tune_parameters(
+      _do_tuning ? getParam<std::vector<std::string>>("tune_parameters")
+                 : std::vector<std::string>{});
```

Comment:
use default in parameter

Reply:
Can not use default because `_do_tuning(isParamValid("tune_parameters"))`

Reply:
ok that works